packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator/test/Providers/NamedTypeSymbolProviders/ConstructorInitializerTests.cs

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.TypeSpec.Generator.Primitives;
+using Microsoft.TypeSpec.Generator.Providers;
+using Microsoft.TypeSpec.Generator.Tests.Common;
+using NUnit.Framework;
+
+namespace Microsoft.TypeSpec.Generator.Tests.Providers.NamedTypeSymbolProviders
+{
+    public class ConstructorInitializerTests
+    {
+        [SetUp]
+        public void Setup()
+        {
+            MockHelpers.LoadMockGenerator();
+        }
+
+        private static Compilation CreateCompilationFromSource(string source)
+        {
+            var syntaxTree = CSharpSyntaxTree.ParseText(source);
+            return CSharpCompilation.Create(
+                assemblyName: "TestAssembly",
+                syntaxTrees: [syntaxTree],
+                references: [MetadataReference.CreateFromFile(typeof(object).Assembly.Location)]);
+        }
+
+        [Test]
+        public void CustomConstructorWithThisInitializerShouldHaveInitializerPopulated()
+        {
+            // Arrange
+            var customCode = @"
+using System;
```

Comment:
We shouldn't use inline code. Instead, we should define custom code following the pattern in https://github.com/microsoft/typespec/blob/main/packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator/test/Providers/ModelProviders/ModelCustomizationTests.cs#L20. The test class is defined under the corresponding TestData folder.

Reply:
Apply this feedback to all added tests.